src/DynamoApplications/StartupUtils.cs

```diff
@@ -0,0 +1,302 @@
+
+            //check for incompatabile parameters
+            if ((!string.IsNullOrEmpty(presetStateid) || (!string.IsNullOrEmpty(presetFile) || (!string.IsNullOrEmpty(verbose)))) && string.IsNullOrEmpty(openfilepath))
+            {
+                Console.WriteLine("you must supply a file to open if you want to load a preset or presetFile, or want to save an evaluation output ");
+                
+            }
+
+
+            return new CommandLineArguments
+            {
+                Locale = locale,
+                CommandFilePath = commandFilePath,
+                OpenFilePath = openfilepath,
+                PresetStateID = presetStateid,
+                PresetFilePath = presetFile,
+                Verbose = verbose,
+            };
+        }
+
+        public string Locale { get; set; }
+        public string CommandFilePath { get; set; }
+        public string OpenFilePath { get; set; }
+        public string PresetStateID { get; set; }
+        public string PresetFilePath { get; set; }
+        public string Verbose { get; set; }
+    }
+
+    public class Preloading
+    {
+        public static void PreloadShapeManager(ref string geometryFactoryPath, ref string preloaderLocation)
+        {
+            var exePath = Assembly.GetExecutingAssembly().Location;
+            var rootFolder = Path.GetDirectoryName(exePath);
+
+            var versions = new[]
+            {
+                LibraryVersion.Version219,
+                LibraryVersion.Version220,
+                LibraryVersion.Version221
+            };
+
+            var preloader = new Preloader(rootFolder, versions);
+            preloader.Preload();
+            geometryFactoryPath = preloader.GeometryFactoryPath;
+            preloaderLocation = preloader.PreloaderLocation;
+        }
+
+        public static DynamoModel MakeModel(bool CLI, IEnumerable<string> librariesToPreload)
+        {
+
+            var geometryFactoryPath = string.Empty;
+            var preloaderLocation = string.Empty;
+            PreloadShapeManager(ref geometryFactoryPath, ref preloaderLocation);
+
+            var config = new DynamoModel.DefaultStartConfiguration()
+                  {
+                      PathResolver = new PathResolver(preloaderLocation, librariesToPreload),
+                      GeometryFactoryPath = geometryFactoryPath,
+                      StartInTestMode = true
+                  };
+
+            //if we are building a model for CLI mode, then we don't want to start an updateManager
+            //for now, building an updatemanager instance requires finding Dynamo install location
+            //which if we are running on mac os or *nix will use different logic then SandboxLookup 
+
+            UpdateManagerConfiguration umConfig = null;
+            if (!CLI)
+            {
+                umConfig = UpdateManagerConfiguration.GetSettings(new SandboxLookUp());
+                Debug.Assert(umConfig.DynamoLookUp != null);
```

Comment:
Is this your `Debug.Assert`? What action do we take if this condition is hit in the application? I'm not a fan of `Debug.Assert` because it just tells you something is wrong without actually controlling the behavior of the application.